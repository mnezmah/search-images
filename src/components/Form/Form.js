import React from 'react'

export default function Form(props) {
  return (
    <div>
      <form onSubmit={props.onsubmit}>
        <input
          type="search"
          name="search"
          placeholder="Search 1.000.000 images..."
          value={props.value}
          onChange={props.onChangeHandler} />
      </form>
    </div>
  )
}
