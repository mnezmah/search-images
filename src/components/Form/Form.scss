@import "../../Styling/main.scss";

form {
  margin: $main-spacing * 2 auto $main-spacing * 2 auto;
  
  //desktop
  // margin: 4%;

  .form__search {
    padding: $main-spacing;
    border-radius: 5px;
    color: $main-background;
    border: none;
    width: 90vw;

    font-size: 1rem;

    //desktop
    // width: 60vw;
    // font-size: 2rem;
    // padding-left: 4%;
  }

  ::placeholder {
    color: $main-grey;
  }
}
