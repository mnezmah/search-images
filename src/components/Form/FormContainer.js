import React, { Component } from 'react'
import { connect } from 'react-redux'
import Form from './Form'
import { getImages } from '../../actions/images'

class FormContainer extends Component {
  state = {
    value: ''
  }

  onSubmit = (e) => {
    e.preventDefault()
    this.props.getImages(this.state.value)
    this.setState({
      value: ''
    })
  }

  onChangeHandler = (e) => {
    this.setState({
      value: e.target.value
    })
  }

  render() {
    return (
      <div>
        <Form value={this.state.value}
          onChangeHandler={this.onChangeHandler}
          onsubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    images: state.images
  }
}

const mapDispatchToProps = {
  getImages
}

export default connect(mapStateToProps, mapDispatchToProps)(FormContainer)