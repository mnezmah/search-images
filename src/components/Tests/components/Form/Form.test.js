import React from 'react'
import { shallow } from 'enzyme'
import Form from '../../../Form/Form'

test('should render Form corectly', () => {
  const wrapper = shallow(<Form />)
  expect(wrapper).toMatchSnapshot()
})