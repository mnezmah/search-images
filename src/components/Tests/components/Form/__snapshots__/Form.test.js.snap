// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Form corectly 1`] = `ShallowWrapper {}`;
